routes/api/contacts.js

```diff
@@ -1,25 +1,82 @@
-const express = require('express')
+const contactsMethots = require("./contactsMethots.js");
+const express = require("express");
+const router = express.Router();
+const Joi = require("joi");
 
-const router = express.Router()
+const schema = Joi.object({
+  name: Joi.string().alphanum().min(3).max(30).required(),
+  email: Joi.string().email({
+    minDomainSegments: 2,
+    tlds: { allow: ["com", "net"] },
+  }),
+  phone: Joi.string().alphanum().min(7).max(12).required(),
+}).with("username", "birth_year");
 
-router.get('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/", async (req, res, next) => {
```

Comment:
Chyba średnio to wytłumaczyłem ostatnio przy tym postcie (post już jest ok, ale chciał bym abyś przerzuciła te try/catche) z contact methods tutaj. Możliwe że ten await jest skomplikowany ale on przeciąga błędy do funkcji nadrzędnej.

```
 router.get("/", async (req, res, next) => {
  try {
      const contacts = await contactsMethots.listContacts();
      res.status(200).json(contacts);
  } catch (error) {
     console.error(error);
     res.status(500).json({  message: "Unknown error occurred" });
  } 
});

```

```
async function listContacts() {
    const data = await fs.readFile(contactsPath, "utf-8");
    return JSON.parse(data);
}
```